oidc/provider.go

```diff
@@ -660,3 +663,113 @@ func (p *Provider) validRedirect(uri string) error {
 	}
 	return fmt.Errorf("%s: redirect URI %s: %w", op, uri, ErrUnauthorizedRedirectURI)
 }
+
+// DiscoveryInfo is the Provider's configuration info which is published to
+// a well-known discoverable location (based on the Issuer of the provider).
+type DiscoveryInfo struct {
+	// Issuer (REQUIRED): Issuer is a case-sensitive URL string using the https
+	// scheme that contains scheme, host, and optionally, port number and path
+	// components and no query or fragment components.
+	Issuer string `json:"issuer"`
+
+	// AuthURL (REQUIRED): URL of the OP's OAuth 2.0 Authorization Endpoint
+	AuthURL string `json:"authorization_endpoint"`
+
+	// TokenURL (REQUIRED): URL of the OP's OAuth 2.0 Token Endpoint
+	TokenURL string `json:"token_endpoint"`
+
+	// UserInfoURL (OPTIONAL, omitempty): URL of the OP's UserInfo Endpoint
+	UserInfoURL string `json:"userinfo_endpoint,omitempty"`
+
+	// JWKSURL (REQUIRED): URL of the OP's JSON Web Key Set [JWK] document
+	JWKSURL string `json:"jwks_uri"`
+
+	// ScopesSupported (RECOMMENDED, omitempty): scope values that this server
+	// supports.
+	ScopesSupported []string `json:"scopes_supported,omitempty"`
+
+	// GrantTypesSupported (OPTIONAL, omitempty):  a list of the OAuth 2.0
+	// Grant Type values that this OP supports. Dynamic OpenID Providers
+	// MUST support the authorization_code and implicit Grant Type values
+	// and MAY support other Grant Types. If omitted, the default value is
+	// ["authorization_code", "implicit"].
+	GrantTypesSupported []string `json:"grant_types_supported,omitempty"`
+
+	// IdTokenSigningAlgsSupported (REQUIRED): a list of the JWS signing
+	// algorithms (alg values) supported by the OP for the ID Token to
+	// encode the Claims in a JWT [JWT]. The algorithm RS256 MUST be included.
+	IdTokenSigningAlgsSupported []string `json:"id_token_signing_alg_values_supported"`
+
+	// DisplayValuesSupported (OPTIONAL, omitempty): a list of the display parameter
+	// values that the OpenID Provider supports.
+	DisplayValuesSupported []string `json:"display_values_supported,omitempty"`
+
+	// UILocalesSupported (OPTIONAL, omitempty): Languages and scripts supported
+	// for the user interface
+	UILocalesSupported []string `json:"ui_locales_supported,,omitempty"`
```

Comment:
Looks like an additional comma snuck in before `omitempty`?

Reply:
ty!